test/irqbalance.bats

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env bats
+# vim: set syntax=sh:
+
+load helpers
+
+function detect_irqbalance_config() {
+	# debian/ubuntu
+	[ -f /etc/default/irqbalance ] && echo "/etc/default/irqbalance"
+	# fedora/centos/RHEL
+	[ -f /etc/sysconfig/irqbalance ] && echo "/etc/sysconfig/irqbalance"
+	# default
+	echo ""
+}
+
+function setup_file() {
+	if ! command -v irqbalance > /dev/null; then
+		skip "irqbalance not found."
+	fi
+	IRQBALANCE_CONF=$(detect_irqbalance_config)
+	if [ -z "$IRQBALANCE_CONF" ]; then
+		echo "error: unable to find irqbalance config file"
+		return 1
+	fi
+	CONFIGLET="$CRIO_CONFIG_DIR/99-irqbalance.conf"
+
+	mkdir -p "/etc/sysconfig"
+	BANNEDCPUS_CONF="/etc/sysconfig/orig_irq_banned_cpus"
+
+	export IRQBALANCE_CONF
+	export CONFIGLET
+	export BANNEDCPUS_CONF
+}
+
+function setup() {
+	setup_test
+	stop_crio
+	# we don't unconditionally restore because the irqbalance package may be missing
+	# from the test system. if this is the case, one less thing to worry about, we can't
+	# pollute the system state anyway :)
+	if [ -f "$IRQBALANCE_CONF" ]; then
+		cp -v "$IRQBALANCE_CONF" "$IRQBALANCE_CONF".bkp
+		expected_banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+		if [ -z "$expected_banned_cpus" ]; then
+			echo "warning: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}. Creating default value:0"
+			echo "IRQBALANCE_BANNED_CPUS=\"0\"" > "$IRQBALANCE_CONF"
+		fi
+	else
+		# placeholder to make the rest of the suite simpler.
+		echo "IRQBALANCE_BANNED_CPUS=\"0\"" > "$IRQBALANCE_CONF"
+		touch /tmp/.test_owns_irqbalance_conf
+	fi
+	if [ -f "$BANNEDCPUS_CONF" ]; then
+		cp -v "$BANNEDCPUS_CONF" "$BANNEDCPUS_CONF".bkp
+	else
+		# empty file is fine and meaningful
+		touch "$BANNEDCPUS_CONF"
+		touch /tmp/.test_owns_bannedcpus_conf
+	fi
+}
+
+function teardown() {
+	cleanup_test
+	stop_crio
+	# see setup about why we have these conditionals
+	if [ -f "$IRQBALANCE_CONF".bkp ]; then
+		mv -v "$IRQBALANCE_CONF".bkp "$IRQBALANCE_CONF"
+	elif [ -f /tmp/.test_owns_irqbalance_conf ]; then
+		rm -f "$IRQBALANCE_CONF"
+		rm -f /tmp/.test_owns_irqbalance_conf
+	else
+		echo "error: Something went really wrong with $IRQBALANCE_CONF !"
```

Comment:
let's try to make the error message more informative and actionable. This is a bit too vague